sda/cmd/api/api.go

```diff
@@ -202,10 +203,26 @@ func checkDB(db *database.SDAdb, timeout time.Duration) error {
 	return db.DB.PingContext(ctx)
 }
 
+func auditLog(token jwt.Token, c *gin.Context) {
+	var bodyBytes []byte
+	bodyBytes, _ = io.ReadAll(c.Request.Body)
+	// Reading the request body also consumes it, this restores the request to the original state before read
+	c.Request.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+	log.WithFields(log.Fields{
+		"audit":  true,
+		"user":   token.Subject(),
+		"method": c.Request.Method,
+		"path":   c.Request.URL.Path,
+		"route":  c.FullPath(),
+		"body":   string(bodyBytes),
+	}).Info("incoming request")
+}
+
```

Comment:
```suggestion
func auditLog(token jwt.Token, c *gin.Context) {
	log.WithFields(log.Fields{
		"audit":  true,
		"user":   token.Subject(),
		"path":   c.Request.URL.Path,
	}).Info("incoming request")
}

```

This should probably be enough